### PR TITLE
chore-xxx: added interceptor

### PR DESCRIPTION
**Pull request by ds-ashanmugavel**

### CODE REVIEW

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/IrsClientRestTemplateInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+
+
+@RequiredArgsConstructor
+@Slf4j
+public class IrsClientRestTemplateInterceptor implements ClientHttpRequestInterceptor {
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/IrsClientRestTemplateInterceptor.java:43]:
```suggestion
public class RequestResponseLoggingInterceptor implements ClientHttpRequestInterceptor {
```

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/RequestResponseLoggingInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+
+
+@RequiredArgsConstructor
+@Slf4j
+public class RequestResponseLoggingInterceptor implements ClientHttpRequestInterceptor {
+
+    static Logger LOGGER = LoggerFactory.getLogger(RequestResponseLoggingInterceptor.class);
```
Comment (ds-mwesener) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/RequestResponseLoggingInterceptor.java:45]: Remove this line.

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/RequestResponseLoggingInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        BufferedReader br = null;
+        try {
+            String requestBody = new String(body, StandardCharsets.UTF_8);
+            LOGGER.info("Request body: {}", requestBody.replaceAll("[\r\n]+", " "));
+            isr = new InputStreamReader(
+                    response.getBody(), StandardCharsets.UTF_8);
+            br = new BufferedReader(isr);
+            String bodys = br.lines()
+                    .collect(Collectors.joining("\n"));
+            LOGGER.info("Response body: {}", bodys.replaceAll("[\r\n]+", " "));
```
Comment (ds-mwesener) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/RequestResponseLoggingInterceptor.java:62]: simply use log.

**Review by ds-mwesener — CHANGES_REQUESTED** [review]
Fix mentioned things.

**Review by ds-mwesener — APPROVED** [review]

**Pull request merged**